CONTRIBUTING.md

```diff
@@ -33,109 +41,210 @@ For an introduction to the basics of reST syntax, the source format in which Spy
+### Create and activate a fresh environment
 
+We highly recommend you create and activate a virtual environment to avoid any conflicts with other packages on your system or causing any other issues.
+Of course, you're free to use any environment management tool of your choice (conda, virtualenvwrapper, pyenv, etc).
+Regardless of the tool you use, make sure to remember to always activate your environment before using it.
 
-### Install dependencies
 
-Make sure you have the appropriate dependencies installed in your active Python environment for the script to work.
-You can install them into your current conda environment with:
+#### Conda
+
+To do so with Conda (recommended), simply execute the following:
+
+```shell
+conda create -c conda-forge -n spyder-docs-env python=3.9
+```
+
+And activate it with
+
+```shell
+conda activate spyder-docs-env
+```
+
+
+#### Pip/Venv
+
+With pip/venv, you can create a virtual environment with
+
+```shell
+python -m venv spyder-docs-env
+```
+
+And activate it with the following on Linux and macOS,
 
 ```bash
+source spyder-docs-env/bin/activate
+```
+
+or on Windows (``cmd.exe``),
+
+```cmd
+.\spyder-docs-env\Scripts\activate.bat
+```
+
+
+
+### Install dependencies
+
+Then, you need to install the appropriate dependencies in your active Python environment to develop and build the documentation.
+You can install them into your current Conda environment with:
+
+```shell
 conda install -c conda-forge --file requirements-conda.txt
-pip install -r requirements-dev.txt
+python -m pip install -r requirements-dev.txt
 ```
 
-Or, if using ``pip`` (``pip3`` on Linux), you can grab them with just:
+Or, if using ``pip``, you can grab them with just:
 
-```bash
-pip install -r requirements-dev.txt
+```shell
+python -m pip install -r requirements-dev.txt
 ```
 
-### Set up Pre-Commit
 
-This repository uses [pre-commit](https://pre-commit.com/) to install, configure and update a suite of pre-commit hooks that check for common problems and issues and fix many of them automatically.
-Pre-commit itself is installed with the above command, and the hooks can be installed by running the following from the root of this repo:
+### Install the required Pre-Commit hooks
 
-```bash
-pre-commit install
-pre-commit install --hook-type commit-msg
+This repository uses [Pre-Commit](https://pre-commit.com/) to install, configure and update a suite of pre-commit hooks that check for common problems and issues, and fix many of them automatically.
+You'll need to install the pre-commit hooks before committing any changes, as they both auto-generate/update specific files and run a comprehensive series of checks to help you find likely errors and enforce the project's code quality guidelines and style guide; they are also run in CI, and will fail the build if any don't pass or modify any files.
+Pre-commit itself is installed with the above command, and the hooks should be enabled by running the following from the root of this repo:
```

Comment:
```suggestion
Pre-commit itself is installed with the [above command](#install-dependencies), and its hooks should be enabled by running the following from the root of this repo:
```

* To me it's not clear what command you're talking about here, but I think it's the one from the previous section, right?
* Second suggestion makes the sentence clearer to me.

Reply:
Great call to reference the specific command, I agree it sounds a bit clearer. Just one fix to my original language ("command" -> "commands") since there are multiple commands there (my mistake originally).

```suggestion
Pre-commit itself is installed with the [above commands](#install-dependencies), and its hooks should be enabled by running the following from the root of this repo:
```